### ISSUE TITLE
Regression regarding bibliography and externally defined terms

### ISSUE THREAD

**Issue by cconcolato** [tabatkins/bikeshed#2363]
There seems to be a regression. My spec had something like:
```<pre class='biblio'>
{
  "AV1": {
  "href": "https://aomediacodec.github.io/av1-spec/av1-spec.pdf",
  "id": "AV1",
  "title": "AV1 Bitstream & Decoding Process Specification",
  "status": "Standard",
  "publisher": "AOM"
    }
</pre>
<pre class="anchors">
url: #biblio-av1; spec: AV1; type: dfn;
    text: AV1 coded video sequence
</pre>
```
It used to work but stopped working see https://github.com/AOMediaCodec/av1-hdr10plus/issues/47#issuecomment-1275124006
The current workaround is to replace `#biblio-av1` with the `href` value.

**Comment by tidoust** [comment]
I suspect that's related to #2364. Thanks in advance for not running a `git blame` to identify the root cause ;)

**Comment by tabatkins** [comment]
Nope, this was actually a different issue altogether. @cconcolato, a <pre class=anchors> is expected to contain external references (all your local refs are, after all, already in your document), so when I refactored the code that generates dfn panels it ended up depending on that assumption more strongly, but your spec's anchor block defines a bunch of things with purely local URLs.

I've fixed this now; the code doesn't crash in these cases, but also it detects a local link in your anchors block and warns about it. So your "workaround" is in fact mostly correct; the only thing is that you probably want to use `urlPrefix` for most of these so they'll generate a proper URL with a correct fragment. (The PDFs won't have a valid fragment, but switching to urlPrefix will at least give them all a unique URL, so the dfn panels will generate correctly; right now all of the refs with the same url coalesce into a single dfn panel, and the first such definition i happen to see claims that panel, so the rest get nothing.)
